### ISSUE TITLE
[Feature Request]: Possibility to choose a Memory Card for each game.

### ISSUE THREAD

**Issue by Sonny35** [PCSX2/pcsx2#8417]
### Description

In 'Settings -> Game Properties' there is no section for the Memory Card. It would be interesting to be able to choose which Memory Card to use for which game.

### Reason

This would be a way to organise your game saves. A Memory Card could be created in a folder format for each game with the name of the game itself, so that it would be more visual to find the specific save game.

Another option for organisation would be a Memory Card in folder format for games you have not yet completed and another one for games you have already completed.

It would be a way to keep both games independent and without having to change the Memory Card every time you switch games. The desired organisation would be up to each user.

### Examples

**COMPLETE/INCOMPLETE GAMES OPTION:**
Memory Card 1: (As global setting)
- DBZ Budokai 3

Memory Card 2: (Specific for completed games)
- DBZ Budokai
- DBZ Budokai 2

**MEMORY CARD OPTION FOR EACH GAME:**
Memory Card 1 called 'DBZ Budokai'.
- DBZ Budokai

Memory Card 1 named 'DBZ Budokai 2'.
- DBZ Budokai 2

Memory Card 1 named 'DBZ Budokai 3'.
- DBZ Budokai 3

**Comment by Mrlinkwii** [comment]
why not try folder memcards ? because they do as you describe

**Comment by Sonny35** [comment]
I already use that option, the problem is that the folders are saved with the game ID and it is not intuitive to search for the save game you want.

**Comment by refractionpcsx2** [comment]
why do you need to search for a save game? the saves are selected automatically depending on the game you load.

**Comment by Sonny35** [comment]
Because I move the game saves between the emulator and the console depending on where I want to play or depending on the compatibility/performance of the emulator with some games, among others.

Also, as I said, it would allow to be able to sort them in different folders for example if you want to have separated the completed games from the games you are still playing without having to change the global configuration every time you switch between those games.

**Comment by refractionpcsx2** [comment]
Surely you have the same issue going back from the console to PCSX2, no? so you should know what folder you need from that.

**Comment by refractionpcsx2** [comment]
The only other solution would be per-game memory card settings, not sure how that would work right now.

**Comment by Sonny35** [comment]
Yes, but on the physical console I have no other solution than to organise the games in two different Memory Cards and swap them, which already filters the saved games depending on the Memory Card I use. On the PCSX2 side it would save work.

Also, many other options allow you to change them specifically for the game, this would be a change that would be in resonance with the rest of the features.

Obviously I know it might not be a simple thing to implement, but many users like me would appreciate this feature. I hope you can keep it in mind even if it's in the long term.

**Comment by Sonny35** [comment]
Yes, but on the physical console I have no other solution than to organise the games in two different Memory Cards and swap them, which already filters the saved games depending on the Memory Card I use. On the PCSX2 side it would save work.

Also, many other options allow you to change them specifically for the game, this would be a change that would be in resonance with the rest of the features.

Obviously I know it might not be a simple thing to implement, but many users like me would appreciate this feature. I hope you can keep it in mind even if it's in the long term.

**Comment by refractionpcsx2** [comment]
Well I'm not going to suggest it for folder memcards, despite those being the easiest to transfer them off, because of how they work, the names need to be as they are.  

Per game memcards would be the simpler solution.

**Comment by Sonny35** [comment]
In case it helps, another possible solution for folder-type Memory Cards would be to be able to make subfolders, in the same way as with ISOs. The game could recursively search for the game saved in the folders of the folder type Memory Card.

Thanks for all your help!

**Comment by RedPanda4552** [comment]
The directory name is at the discretion of the game itself, not the console or an emulator. If a game is picking an unfavorable directory name that is an issue with the game and not fixable without patching.

Folder memory cards already isolate non-relevant game saves from the running game, so making additional memory cards is just creating clutter and wasting disk space.

**Comment by Sonny35** [comment]
Well, I guess that depends on each user and how they want to organise themselves. The change I'm suggesting shouldn't change anything to how games are used today, it's just adding new functionality in a way that doesn't interfere with what's already in use. If you prefer to continue using it as it is now, it wouldn't change anything.

**Comment by RedDevilus** [comment]
Think Duckstation has per-game memory card save files. But folder memcard is pretty close to it (minus the memcard filter nonsense)

**Comment by Blackbird88** [comment]
For now you can do this.
https://github.com/PCSX2/pcsx2/issues/6591#issuecomment-1283440268

**Comment by Sonny35** [comment]
> For now you can do this. [#6591 (comment)](https://github.com/PCSX2/pcsx2/issues/6591#issuecomment-1283440268)

It worked completely with both folder Memory Cards and archive Memory Cards. @refractionpcsx2 So actually the option is already programmed, seems that you just need to add it in the Game Properties menu.

**Comment by Scall** [comment]
It should also be noted that the downside of using folder memory cards is currently the worse compatibility: https://github.com/PCSX2/pcsx2/issues/3918

I think if PCSX2 wrote per-game 8MB file memory cards compressed (e.g., with **libzip**), all the space issues would be solved for everyone while retaining maximum compatibility.

**Comment by RedDevilus** [comment]
> It should also be noted that the downside of using folder memory cards is currently the worse compatibility: #3918
> 
> I think if PCSX2 wrote per-game 8MB file memory cards compressed (e.g., with **libzip**), all the space issues would be solved for everyone while retaining maximum compatibility.

Some games easily fill up several megabytes for a single save, and what about people that save spam? I see your point though, but it might be easier to improve folder memcards to expose everything when in BIOS and not needing memcard filters. Not saying it is impossible, savestates are compressed and faster than before, nothing preventing doing that on file memcards.

Per-game detection for memcards might be a more niche, some frontends do it too, I remember PCSX2 configurator plugin on launchbox did it before it died due to not being worked on anymore.

**Comment by MageKing17** [comment]
Actually, if you switch to Big Picture Mode, individual game properties already have a Memory Card Settings submenu:
![image](https://user-images.githubusercontent.com/2095329/230747495-1d0dcea5-baf5-4eb0-b654-4988fdae26c8.png)

**Comment by RedDevilus** [comment]
They want a memcard per game, which is not exactly the same as folder memcard. Arguably it's kinda there in a different way people expect it to be.

Folder memcard: 
- game1
- game2
- game3


File memcard:
- game1

Per-game (file) memcard:
Game1.ps2
- game1
Game2.ps2
- game2
Game3.ps2
- game3

**Comment by HeyItsLollie** [comment]
I'm another user who wants a per-game setting for memory cards. I want to be able to create and select a memory card for a specific game (as a per-game override), in a location outside of PCSX2's default locations. eg: Dropbox.

My use case is a little niche: A friend and I play single-player games together over Discord, and we switch whoever's playing/streaming. This requires us to have access to the same memory card - we can always pass the card file back and forth manually if we need to, but being able to automate this via Dropbox really simplifies things.

(We use "Save State to File" as well, but it's good to have the memory card for stability.)

We were able to achieve this with DuckStation, which effectively allows for custom per-game memory cards via its "Shared Between All Games" option. We just create a new card in a shared folder in Dropbox, and point the game settings towards this file.

![2023-05-21 - 11-21-29](https://github.com/PCSX2/pcsx2/assets/30498581/a33de87e-6035-4802-8007-5f0153f4b834)

This is basically what I'd like to see in PCSX2, just an override to specify which memory cards are used for a game.
